pkg/feature/feature_tracker_handler.go

```diff
@@ -50,39 +56,36 @@ func (f *Feature) createFeatureTracker(ctx context.Context) error {
 }
 
 func removeFeatureTracker(ctx context.Context, f *Feature) error {
-	if err := getFeatureTrackerIfAbsent(ctx, f); err != nil {
-		return client.IgnoreNotFound(err)
+	associatedTracker := f.Tracker
+	if associatedTracker == nil {
+		// Check if it is persisted in the cluster, but Feature do not have it attached
+		if tracker, errGet := getFeatureTracker(ctx, f); client.IgnoreNotFound(errGet) != nil {
+			return errGet
+		} else {
+			associatedTracker = tracker
```

Comment:
is there any possibility that after `associatedTracker = tracker` associatedTracker is nil? 
otherwise why not just
`return client.IgnoreNotFound(f.Client.Delete(ctx, tracker))`

Reply:
In theory, it can be not found in L62, and so it is `nil`. We do not propagate error on 404 (`client.IgnoreNotFound(errGet)`), but we handle this case by simply skipping the Delete operation. Calling `Delete` on `nil` will panic.

As to part if this can happen at runtime - under the hood it boils down to which operations are performed directly against `kube-apiserver` and which are taken from the local cache. I think it can be stale and might lead to such situation, so I thought it's better to handle it explicitly.